### PR TITLE
Create interop procedure

### PR DESCRIPTION
**Pull request by jrhea**
First cut at the initial interop procedure

### CODE REVIEW

File: hobbits-client_interop_proceedure.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+### Procedure
+
+For the initial round: 
+1. Clients will pair off in teams of two.
+2. `Eth2Genesis` will be triggered on startup
+3. Once the clients can finalize the chain, then they pick another client pair to interop with.  
+
+Repeat (1) & (2) until all clients are talking.
+
+>**Round 1:** 2 nodes (one per client) with 16 validators/client
```
Comment (prestonvanloon) [inline hobbits-client_interop_proceedure.md]: Formatting here is messed up

Comment (jrhea) [reply hobbits-client_interop_proceedure.md]: gracias

File: interop/hobbits-interop_procedure.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
+# Initial Interop
```
[the remark this answers is not in the document]
Comment (jrhea) [reply interop/hobbits-interop_procedure.md]: Afk right now. Will do when I get home. Thanks for the review

[the remark this answers is not in the document]
Comment (jrhea) [reply interop/hobbits-interop_procedure.md]: ok this should be resolved now

**Comment by jrhea** [comment]
@hwwhww I made the mods you suggested.  lmk if there is anything else

File: interop/hobbits/README.md [hunk 3]
```diff
@@ -0,0 +1,71 @@
+# Initial Interop
+
+For the initial client interop, clients will utilize a simplified networking implementation and wire protocol. For more information please review the hobbits specification [here](https://github.com/deltap2p/hobbits).
+
+### Prereqs
+
+- targeting the frozen spec (v0.8)
+- all clients must pass reference tests
+- all clients must pass [hobbits integration tests](./hobbits-integration-tests.md)
```
Comment (djrtwo) [inline interop/hobbits/README.md]: This link is broken

```suggestion
- all clients must pass [hobbits integration tests](./integration-tests.md)
```

File: interop/hobbits/README.md [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+
+- all wire protocol implementations must pass [hobbits conformance tests](https://github.com/jrhea/eth2.0-pm/blob/patch-2/interop/hobbits-conformance-tests.md)
```
Comment (djrtwo) [inline interop/hobbits/README.md]: broken link
```suggestion
- all wire protocol implementations must pass [hobbits conformance tests](./conformance-tests.md)
```

File: interop/hobbits/README.md [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+
+## Stage One: Coordinated Start
+
+Clients simulate a coordinated genesis event and gossip blocks/attestations for each slot.  This stage demonstrates the ability of clients to finalize the chain starting from `Eth2Genesis`.
+
+### Prereqs
+
+- Ability to read in the N number of deposits required to trigger the `Eth2Genesis` log
```
Comment (djrtwo) [inline interop/hobbits/README.md]:
```suggestion
- Ability to read in the N number of deposits required to trigger the `is_genesis_trigger` function
```

File: interop/hobbits/README.md [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+- Ability to gossip blocks
```
Comment (djrtwo) [inline interop/hobbits/README.md]:
```suggestion
- Ability to gossip blocks and attestations
```

File: interop/hobbits/README.md [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+- Static peering
+- TCP transport
+
+### Procedure
+
+For the initial round: 
+1. Clients will pair off in teams of two.
+2. `Eth2Genesis` will be triggered on startup
```
Comment (djrtwo) [inline interop/hobbits/README.md]:
```suggestion
2. `is_genesis_trigger` will be triggered on startup
```

File: interop/hobbits/README.md [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+### Prereqs
+
+- Ability to read in the N number of deposits required to trigger the `Eth2Genesis` log
+- Ability to gossip blocks
+- Static peering
+- TCP transport
+
+### Procedure
+
+For the initial round:
```
Comment (djrtwo) [inline interop/hobbits/README.md]: We should include the [constant config](https://github.com/ethereum/eth2.0-specs/tree/master/configs/constant_presets) to be used here. I recommend either "minimal" or something similarly small. We can add an even more minimal one to that set if we think valuable, but minimal should work well

File: interop/hobbits/README.md [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+
+Repeat (1) & (2) until all clients are talking.
+
+>**Round 1:** 2 nodes (one per client) with 16 validators/client  
+>**Round 2:** 4 nodes (one per client) with 8 validators/client  
+>**Round 3:** 8 nodes (one per client) with 4 validators/client
+
+### Goal
+
+Clients can declare success at each round after 3 epochs where:
```
Comment (djrtwo) [inline interop/hobbits/README.md]: I would add a "stretch" goal of having run for ~15 minutes without error and continual finalization or something. This seems to be an implied pre-req for "stage two"

File: interop/hobbits/README.md [hunk 10]
```diff
@@ -0,0 +1,71 @@
...
+- RPC methods implemented so that a new client can sync to an existing testnet.
+- Standup a short-lived heterogeneous testnet 
+- Static peering
+- TCP transport
+
+### Procedure
+
+For the initial round: 
+1. Clients will pair off in teams of two  
+2. Both client A & B demonstrate the ability to sync to an existing testnet
```
Comment (djrtwo) [inline interop/hobbits/README.md]: Should we add more details here? Is the point that A and B create testnet, then A adds another node to sync from scratch and B adds another node to sync from scratch? I'm assuming the additional nodes do _not_ have validators?

File: interop/hobbits/integration-tests.md [hunk 11]
```diff
@@ -0,0 +1,19 @@
+# integration testing procedure
+
+![hit](hit.png)
+
+### Test 1: Coordinated Start
+
+1. Both clients are connected as above and started
+2. `Eth2Genesis` will be triggered immediately
```
Comment (djrtwo) [inline interop/hobbits/integration-tests.md]:
```suggestion
2. `is_genesis_trigger` will be triggered immediately
```

File: interop/hobbits/integration-tests.md [hunk 12]
```diff
@@ -0,0 +1,19 @@
...
+3. Messages between clients are routed through the relay
+4. The relay decodes and encodes each message as it passes thru the relay
+5. If the clients are able to justify and finalize, then they pass
+
+### Test 2: Join & Sync
+
+1. Both clients are connected as above and started
+2. Clients run for a couple of epochs
+3. A third client joins the network and asks to sync
```
Comment (djrtwo) [inline interop/hobbits/integration-tests.md]: does third client contain validators or is it a non-validating node?

Comment (jrhea) [reply interop/hobbits/integration-tests.md]: i was think of a non-validating node initially

File: interop/hobbits/README.md (hunk 10 above)
Comment (jrhea) [reply interop/hobbits/README.md]: ya you got the idea.  i can add more words

File: interop/hobbits/README.md (hunk 9 above)
Comment (jrhea) [reply interop/hobbits/README.md]: ya....good idea

File: interop/hobbits/integration-tests.md (hunk 12 above)
Comment (djrtwo) [reply interop/hobbits/integration-tests.md]: makes sense. otherwise, you'd probably want to make sure 1st two clients have >2/3 of the vset

**Comment by jrhea** [comment]
@djrtwo i believe all your comments should be resolved now

**Pull request merged**